CoinDock-api/app/Models/V1/Coin.php

```diff
@@ -4,15 +4,70 @@
 
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Http;
 
 class Coin extends Model
 {
     use HasFactory;
-    protected $fillable = ['name'];
+    protected $fillable = ['coin_id', 'name', 'is_crypto', 'status', 'img_path', 'is_default'];
+
+
+    public function wallets()
+    {
 
-    public function wallet(){
         return $this->hasMany(Wallet::class);
     }
 
-    
+    //Number of Coins
+    public function countCoins()
+    {
+        $user = Auth::user();
+        return $this->wallets()->whereUserId($user->id)
+            ->whereCoinId($this->id)
+            ->sum('coins');
+    }
+
+    //Convertion
+    public function priceConversion($from, $to, $grouped)
+    {
+        $baseUrl = config('coinapi.coin.api_url');
+        $currencyURL = $baseUrl . config('coinapi.coin.exchange_url');
+        $cryptConversionURL = str_replace(['{from}', '{to}'], [$from, $to], $currencyURL);
+        $response = Http::withHeaders(['X-CoinAPI-Key' => config('coinapi.coin.api_key')])->get($cryptConversionURL);
+        $price = $response['rate'];
+        return $grouped * $price;
+    }
+
+    //get the primary currency value
+    public function getPrimaryCurrency()
+    {
+        //$from= 'BTC';
+        $user = Auth::user();
+        $grouped = $this->countCoins($user);
+        $from = Coin::whereIsDefault(1)->first()?->coin_id;
+        $to = Setting::whereUserId($user->id)->first()?->primary_currency;
```

Comment:
Are we have any relation for user model and setting model? if yes then use that relation here instead of 
Setting::whereUserId($user->id)->first()?->primary_currency; query

Reply:
Done